src/Temporalio/Client/EnvConfig/ClientEnvConfig.cs

```diff
@@ -0,0 +1,414 @@
+            /// <returns>The loaded profile.</returns>
+            public static Profile Load(ProfileLoadOptions options)
+            {
+                var runtime = Runtime.TemporalRuntime.Default.Runtime;
+                return Bridge.EnvConfig.LoadClientConfigProfile(runtime, options);
+            }
+
+            /// <summary>
+            /// Create a Profile from a dictionary structure.
+            /// </summary>
+            /// <param name="dictionary">The dictionary to convert from.</param>
+            /// <returns>Profile configuration instance.</returns>
+            public static Profile FromDictionary(IReadOnlyDictionary<string, object?> dictionary) =>
+                new(
+                    Address: dictionary.TryGetValue("address", out var address) ? (string?)address : null,
+                    Namespace: dictionary.TryGetValue("namespace", out var nameSpace) ? (string?)nameSpace : null,
+                    ApiKey: dictionary.TryGetValue("api_key", out var apiKey) ? (string?)apiKey : null,
+                    Tls: dictionary.TryGetValue("tls", out var tls) && tls is IReadOnlyDictionary<string, object?> tlsDict ? Tls.FromDictionary(tlsDict) : null,
+                    GrpcMeta: dictionary.TryGetValue("grpc_meta", out var grpcMeta) ? (IReadOnlyDictionary<string, string>?)grpcMeta : null);
+
+            /// <summary>
+            /// Create a <see cref="TemporalClientConnectOptions"/> from this profile.
+            /// </summary>
+            /// <returns>Connection options for a client.</returns>
+            public TemporalClientConnectOptions ToClientConnectionOptions()
+            {
+                var options = new TemporalClientConnectOptions(Address ?? string.Empty);
+                // Set namespace if provided
+                if (Namespace != null)
+                {
+                    options.Namespace = Namespace;
+                }
+
+                // Set default TLS options if API key exists
+                if (ApiKey != null)
+                {
+                    options.ApiKey = ApiKey;
+                    options.Tls = new TlsOptions();
+                }
+
+                // Override with explicit TLS configuration if provided
+                if (Tls != null)
+                {
+                    options.Tls = Tls.ToTlsOptions();
+                }
+
+                // Add gRPC metadata if present
+                if (GrpcMeta != null)
+                {
+                    options.RpcMetadata = new List<KeyValuePair<string, string>>(GrpcMeta);
+                }
+
+                return options;
+            }
+
+            /// <summary>
+            /// Convert to a dictionary structure that can be used for TOML serialization.
+            /// </summary>
+            /// <returns>Dictionary representation of this profile.</returns>
+            public Dictionary<string, object?> ToDictionary() =>
```

Comment:
```suggestion
            public Dictionary<string, object> ToDictionary() =>
```

Not sure any of these values in these dictionaries should be nullable. The key should be absent of the value is null IMO.